Disable Mac OS specific features

We only support Linux so we don't need them. Also, they cause a build
failure when cross compiling:

checking for Mach-O dynamic module iteration functions
error: cannot run test program while cross compiling

Signed-off-by: Vicente Olivert Riera <Vincent.Riera@imgtec.com>

diff -Nrup a/configure.ac b/configure.ac
--- a/configure.ac	2015-09-28 10:33:39.175048493 +0100
+++ b/configure.ac	2015-09-28 10:47:53.921428143 +0100
@@ -467,9 +467,6 @@ fi
 
 dnl Mac OS specific features -------------------
 
-SVN_LIB_MACHO_ITERATE
-SVN_LIB_MACOS_PLIST
-SVN_LIB_MACOS_KEYCHAIN
 
 dnl APR_HAS_DSO -------------------
 
